docs/30DaysOfPWA/advanced-capabilities/06.md

```diff
@@ -1,15 +1,174 @@
+
+Background Sync API is here to help. It allows developers to think about frontend app <-> server communication as about a set of _syncs_. These syncs will happen immediately after they were created ("registered" in terms of Background Sync API) if the connection is available or, if there is no connection, later when the user goes back online. What differs this way from all other pre-PWA approaches is: this "later" will also work even _after_ the user closed the application tab and/or the visible part of the browser - thanks to the service worker that is always "on duty" in the background.
+
+Please note, this API only provides a `sync` event to the service worker. You have to implement preserving the data you want to send (in case of delayed sending because of offline) and sending the data itself. You might need to think about where to store the data (IndexedDB is a good option) and about organizing queuing if you want to support multiple sends within one sync.
+
+There are lots of scenarios for background sync but the most straightforward and widespread one is the following: repeating requests to the server done offline (i.e. failed ones) after the connection is restored. Let's see how it works in practice for a web-based social media client.
+
+### How to start
+
+In your main application code, after the user created a new post and clicked "Publish" you check if the request to the server failed. If it did, you create a _sync_ for this attempt. You need an active service worker registration to do that.
+
+```javascript
+async function publishPostOnConnected(post) {
+  const registration = await navigator.serviceWorker.ready;
+  try {
+    // First, you should write a code to save the data you want to send later. It's a good idea to have this function ready for multiple posts published while offline. Using IndexedDB is the best option for storing this kind of data.
+    await savePost(post);
+
+    // Then, the sync registration itself. We give it a name because we can have multiple syncs for various parts of the app functionality:
+    await registration.sync.register('sync-post');
+
+    // Finally, you can inform the user about it:
+    showNotification('Your post will be published automatically right after connection is restored. It is safe to close the app.');
+  } catch {
+    console.error('Background Sync registration failed');
+  }
+}
+```
+
+It's a really good idea to wrap this code by a feature detection:
```

Comment:
We could change this to mention that this is, currently, only supported on Chromium-browsers. And therefore, perhaps remove the mention of Chromium earlier in the intro, and just keep it general up there.